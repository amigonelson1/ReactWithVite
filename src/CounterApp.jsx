import { useState } from 'react';
import PropsType from 'prop-types';

//guia de eventos: 'https://es.reactjs.org/docs/events.html';
// hooks de react: 'https://es.reactjs.org/docs/hooks-intro.html';

function CounterApp({ value }) {

    const [counter, setCounter] = useState(value)

    const handledAdd = () => {
        setCounter(counter + 1);
        // para cambiar varias veces en un mismo evento seteamos con una funcion interna;
        /*
        setCounter(counter=> counter+1);
        setCounter(counter=> counter+1);
        */
    };

    const handledsubstract = () => {
        setCounter(counter - 1);
    };

    const handledReset = () => {
        setCounter(value);
    };

    return (
        <>
            <h1>CounterApp</h1>
            <h2>{counter}</h2>
            <button onClick={handledAdd}>+1</button>
            <button aria-label='reset-btn' onClick={handledsubstract}>-1</button>
            <button onClick={handledReset}>Reset</button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropsType.number.isRequired,
}

CounterApp.defaultProps = {
    value: 0,
}

export default CounterApp