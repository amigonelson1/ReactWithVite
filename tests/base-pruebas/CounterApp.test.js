import { fireEvent, render, screen } from '@testing-library/react';
import CounterApp from '../../src/CounterApp';

describe('prueba a counter app', () => {

    const valorInicial = 100;

    test('prueba snapshot', () => {
        const { container } = render(<CounterApp value={valorInicial} />);
        expect(container).toMatchSnapshot();
    });

    test('prueba valor inicial de 100', () => {
        render(<CounterApp value={valorInicial} />);
        expect(screen.getByText(valorInicial)).toBeTruthy();
    });

    //prueba para accion clic en botones por nombre visible en pantalla;
    test('incrementa valor al dar clic en +1', () => {
        render(<CounterApp value={valorInicial} />);
        fireEvent.click( screen.getByText('+1') );// '+1' es como apareece el boton en pantalla;
        //screen.debug();// muestra un screen en el log;
        expect(screen.getByText('101')).toBeTruthy();
    });

    //prueba para accion clic en botones identificandolo por su aria-label;
    test('Decrementa valor al dar clic en -1', () => {
        render(<CounterApp value={valorInicial} />);
        fireEvent.click(screen.getByRole('button',{name: 'reset-btn'}));
        expect(screen.getByText('99')).toBeTruthy();
    });
});