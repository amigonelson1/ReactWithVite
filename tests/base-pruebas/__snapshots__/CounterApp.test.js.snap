// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba a counter app prueba snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    100
  </h2>
  <button>
    +1
  </button>
  <button
    aria-label="reset-btn"
  >
    -1
  </button>
  <button>
    Reset
  </button>
</div>
`;
